server/pulp/server/lazy/url.py

```diff
@@ -0,0 +1,594 @@
+        """
+        _hash = sha256()
+        _hash.update(policy)
+        return _hash.hexdigest()
+
+    @staticmethod
+    def load(policy):
+        """
+        Load the policy.
+        A policy object is created
+
+        :param policy: A policy dictionary.
+        :type policy: dict
+        :return: The loaded policy object.
+        :rtype: Policy
+        :raise PolicyMalformed: on malformed policy.
+        """
+        # Empty
+        if not policy:
+            raise PolicyMalformed(_('EMPTY'))
+        # Must be dictionary
+        if not isinstance(policy, dict):
+            raise PolicyMalformed(_('Must be <dict>'))
+        # Resource Specified
+        resource = policy.get(Policy.RESOURCE)
+        if not resource:
+            raise PolicyMalformed(_('Resource must be specified'))
+        # Expiration Integer
+        expiration = policy.get(Policy.EXPIRATION)
+        if not isinstance(expiration, int):
+            raise PolicyMalformed(_('Expiration must be integer'))
+        # Extensions Dictionary
+        extensions = policy.get(Policy.EXTENSIONS)
+        if not isinstance(extensions, dict):
+            raise PolicyMalformed(_('Extensions must be dictionary'))
+        # Done
+        policy = Policy(resource, int(expiration))
+        policy.extensions = extensions
+        return policy
+
+    @staticmethod
+    def decode(encoded):
+        """
+        Decode the base64 encoded json policy.
+
+        :param encoded: A base64 encoded json policy.
+        :type encoded: str
+        :return: The decoded policy.
+        :rtype: Policy
+        """
+        policy = Base64.decode(encoded)
+        policy = JSON.decode(policy)
+        policy = Policy.load(policy)
+        return policy
+
+    @staticmethod
+    def validate(key, encoded, signature):
+        """
+        Decode and validate a policy.
+
+        :param key: A public RSA key.
+        :type key: RSA.RSA
+        :param encoded: A base64 encoded json policy.
+        :type encoded: str
+        :param signature: A base64 encoded RSA signature.
+        :type signature: str
+        :return: The validated and decoded policy.
+        :rtype: Policy
+        :raise NotValid: if the signature and policy digest cannot be
+            validated using the public key. Or, that the policy has expired.
+        """
+        try:
+            digest = Policy.digest(encoded)
+            if not key.verify(digest, Base64.decode(signature)):
+                raise PolicyNotAuthenticated()
+            policy = Policy.decode(encoded)
+            if policy.expiration <= time():
+                raise PolicyExpired()
+            return policy
+        except RSA.RSAError:
+            raise PolicyNotAuthenticated()
+
+    def __init__(self, resource, expiration):
+        """
+        :param resource: A resource that is the subject of the policy.
+        :type resource: str
+        :param expiration: The policy expiration (seconds since epoch).
```

Comment:
Same documentation thing @beav noted below. This is probably added to the current system time (which is in the form of seconds since the epoch), but that's not clear here.